Enables Fortify SCA scans in support of SOC 2 compliance efforts

grimlor opened the pull request:
Note: If the Fortify license is not found in the current user's `.lightbend` directory, the Fortify SCA is not hooked into the build and everything behaves as before.

Changes:
Modifies build.sbt to enable creating NSTs for all modules, outputting to individual folders.
Adds script for scanning all such folders.
Adds the output of said scans for auditing purposes.
Adds scan times CSV for analysis.

<!--
  Are there any relevant issues / PRs / mailing lists discussions?
  Please reference them here - but don't use `fixes` notation.
-->
References #31403

Commit:
Modifies `build.sbt` to enable creating NSTs for all modules, outputting to individual folders.
Adds script for scanning all such folders.
Adds the output of said scans for auditing purposes.
Adds scan times CSV for analysis.

Commit:
Updates `fortify-scan.sh` script to create a log.
Formats the timestamp to be parsable.
Reran scans to verify functionality and included the log in commit.
Note: scan took ~86 minutes.

Commit:
Updates to check for fortify license; no license, no NSTs

raboof commented on build.sbt:
```diff
@@ -590,18 +590,40 @@ lazy val serialversionRemoverPlugin =
+
+def fortifySettings(name: String) =
+  Seq(
+    addCompilerPlugin(("com.lightbend" %% "scala-fortify" % "1.0.21").cross(CrossVersion.patch)),
```
(Might be nice to keep the version in `/project/Dependencies.scala`, but not essential)

raboof reviewed (approved):
I'm not too familiar with fortify, but since this change doesn't have any effect unless you have the fortify license configured I'm OK with it.

(I think it would be possible to move this logic out of `build.sbt` into a separate plugin in `/project`, but I'm not sure that really simplifies things, so having it in `build.sbt` seems fine)

Commit:
Minor refactoring to make more readable.
Adds `fortify-scan.sh` to create fpr and pdf outputs.
Adds `runSourceAnalyzer` task to allow running `fortify-scan.sh` with sbt.
Adds fpr and pdf reports for review and prioritization by engineering.

Commit:
Updates to stop committing `fortify-scans/`.
Now gets the lightbend/fortify-scans repo and pushes results there.
Moves Fortify version to where all other plugin version info is.

Commit:
Updates to stop committing `fortify-scans/`.
Now gets the lightbend/fortify-scans repo and pushes results there.
Moves Fortify version to where all other plugin version info is.

Commit:
Adds updates to properly push the Fortify scan results.

patriknw commented on build.sbt:
```diff
@@ -590,17 +594,36 @@ lazy val serialversionRemoverPlugin =
+lazy val doesFortifyLicenseExist: Boolean = {
+  import java.nio.file.Files
+  val username = System.getenv("USER")
+  val fortifyLicense = new File(s"/Users/$username/.lightbend/fortify.license")
```
better to use `System.getProperty("user.home")` for finding the home directory

patriknw commented on fortify-scan.sh:
```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
```
can we place this in the `scripts/` directory where we have some other scripts

patriknw commented on fortify-scan.sh:
```diff
@@ -0,0 +1,48 @@
+git add .
+commitMsg="Scan performed on $(date +"%Y-%m-%d_%T")"
+git commit -m "$commitMsg"
+git push origin
```
Should this really push to the repo? What is the process for running these scripts? Are they supposed to be run by CI or is it manually from a local machine?

grimlor replied:
Right now, it's manual but I am writing this with the idea of automation. Right now, it's manual but I am writing this with the idea of automation, so CI is the goal. 

As for pushing to the repo, I've changed the script to push the results to a private repo, lightbend/fortify-scans under a folder named for the project being scanned. Technically, anyone could have Fortify SCA and run the same scans but I figured it best to not expose our vulnerabilities so directly as to include with the project.

grimlor committed:
Merge branch 'akka:main' into main

Commit:
Changes to support running scans on platforms other than Mac.
Moves `fortify-scan.sh` to `scripts` folder.

Commit:
Resolves merge conflicts.

patriknw reviewed (approved):
LGTM

Commit:
Deletes (moved) script
Removes unused variable
Cleans up old log file before creating a new one for the current scan session

grimlor committed:
Merge pull request #1 from akka/main

Merging from lightbend/akka

Commit:
Merge remote-tracking branch 'origin/main' into fortify-scan

grimlor commented:
I'm not yet a project collaborator so the above checks have to be run by one of y'all. :)

patriknw commented:
There is a formatting issue in build.sbt. Could you run `scalafmtSbt` and commit the changes.

If it's ready after that we can merge it.

Commit:
Updates Fortify to 22.1.
Adds support for easily updating to new Fortify versions.

Commit:
Removes the need to explicitly compile ahead of performing source analysis by adding a command alias that does both.
Renames script to be more descriptive and align with the other refactoring.

Commit:
Fixes format issues

johanandren commented:
Trying to re-trigger validation
